kimchi/js/bindings.js

```diff
@@ -1424,6 +1447,12 @@ var caml_plonk_verification_evals_to_rust = function (x, klass, poly_comm_class,
         return caml_poly_comm_to_rust_poly_comm(poly_comm, poly_comm_class, mk_affine);
     };
 
+    var convert_opt = function (poly_comm_opt) {
```

Comment:
I don't think this does what we want. The JSOO representation of an OCaml `option` is `0` for `None` or `[0, foo]` for `Some foo`. You can see similar being done elsewhere in this file

Reply:
Doesn't it? Don't we want to map `None` to `undefined` and `Some x` to `caml_poly_comm_to_rust_poly_comm(x, ....)`  in this `to_rust` function ?

Reply:
AFAICT, the code here is very similar to `caml_poly_comm_to_rust_poly_comm` where we have the following snippet
```
    var caml_unshifted = poly_comm[1];
    var caml_shifted = poly_comm[2];
    var rust_shifted = undefined;
    if (caml_shifted !== 0) {
        rust_shifted = rust_affine_of_caml_affine(caml_shifted[1], mk_affine);
    }
```